parts/windows/windowsconfigfunc.ps1

```diff
@@ -196,3 +196,133 @@ function Enable-FIPSMode
+        Write-Log "GmsaPackageUrl is not set so skip installing GMSA plugin."
+        return
+    }
+
+    $tempInstallPackageFoler = $env:TEMP
+    $tempPluginZipFile = [Io.path]::Combine($ENV:TEMP, "gmsa.zip")
+
+    Write-Log "Getting the GMSA plugin package"
+    DownloadFileOverHttp -Url $GmsaPackageUrl -DestinationPath $tempPluginZipFile
+    Expand-Archive -Path $tempPluginZipFile -DestinationPath $tempInstallPackageFoler -Force
+    if($LASTEXITCODE) {
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GMSA_EXPAND_ARCHIVE -ErrorMessage "Failed to extract the '$tempPluginZipFile' archive."
+    }
+    Remove-Item -Path $tempPluginZipFile -Force
+
+    $tempInstallPackageFoler = [Io.path]::Combine($tempInstallPackageFoler, "CCGPlugin")
+
+    # Copy the plugin DLL file.
+    Write-Log "Installing the GMSA plugin"
+    Copy-Item -Force -Path "$tempInstallPackageFoler\CCGAKVPlugin.dll" -Destination "${env:SystemRoot}\System32\"
+
+    # Enable the logging manifest.
+    Write-Log "Importing the CCGEvents manifest file"
+    wevtutil.exe im "$tempInstallPackageFoler\CCGEvents.man"
+    if($LASTEXITCODE) {
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GMSA_IMPORT_CCGEVENTS -ErrorMessage "Failed to import the CCGEvents.man manifest file."
+    }
+
+    # Enable the PowerShell privilege to set the registry permissions.
+    Write-Log "Enabling the PowerShell privilege"
+    $enablePrivilegeResponse = Retry-Command -Command "Enable-Privilege" -Args @{Privilege="SeTakeOwnershipPrivilege"} -Retries 5 -RetryDelaySeconds 5
+    if(!$enablePrivilegeResponse) {
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GMSA_ENABLE_POWERSHELL_PRIVILEGE -ErrorMessage "Failed to enable the PowerShell privilege."
+    }
+
+    # Set the registry permissions.
+    Write-Log "Setting GMSA plugin registry permissions"
+    try {
+        $ccgKeyPath = "System\CurrentControlSet\Control\CCG\COMClasses"
+        $owner = [System.Security.Principal.NTAccount]"BUILTIN\Administrators"
+
+        $key = [Microsoft.Win32.Registry]::LocalMachine.OpenSubKey(
+            $ccgKeyPath,
+            [Microsoft.Win32.RegistryKeyPermissionCheck]::ReadWriteSubTree,
+            [System.Security.AccessControl.RegistryRights]::TakeOwnership)
+        $acl = $key.GetAccessControl()
+        $acl.SetOwner($owner)
+        $key.SetAccessControl($acl)
+        
+        $key = [Microsoft.Win32.Registry]::LocalMachine.OpenSubKey(
+            $ccgKeyPath,
+            [Microsoft.Win32.RegistryKeyPermissionCheck]::ReadWriteSubTree,
+            [System.Security.AccessControl.RegistryRights]::ChangePermissions)
+        $acl = $key.GetAccessControl()
+        $rule = New-Object System.Security.AccessControl.RegistryAccessRule(
+            $owner,
+            [System.Security.AccessControl.RegistryRights]::FullControl,
+            [System.Security.AccessControl.AccessControlType]::Allow)
+        $acl.SetAccessRule($rule)
+        $key.SetAccessControl($acl)

```

Comment:
$key.Close()? 
_Closes the key and flushes it to disk if its contents have been modified._

Reply:
It works from my testing. This is only used to run `reg.exe import ` in this process so I think that we do not need to save the data.

Reply:
I guess it's like a file.close(), the decision is given to you.

Reply:
If we need to persistent this change for usage after reboot, we definitely need to call it. But we only use it after enabling privilege and do not want to use it after reboot, I think that it is better not to close it.